dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/CompleteDataSetRegistrationController.java

```diff
@@ -195,42 +193,37 @@ public void getCompleteRegistrationsJson(
     // POST
     // -------------------------------------------------------------------------
 
-    @PostMapping( consumes = CONTENT_TYPE_XML )
-    public void postCompleteRegistrationsXml(
+    @PostMapping( consumes = CONTENT_TYPE_XML, produces = CONTENT_TYPE_XML )
+    @ResponseBody
+    public WebMessage postCompleteRegistrationsXml(
         ImportOptions importOptions, HttpServletRequest request, HttpServletResponse response )
         throws IOException
     {
         if ( importOptions.isAsync() )
         {
-            asyncImport( importOptions, ImportCompleteDataSetRegistrationsTask.FORMAT_XML, request, response );
-        }
-        else
-        {
-            response.setContentType( CONTENT_TYPE_XML );
-            ImportSummary summary = registrationExchangeService
-                .saveCompleteDataSetRegistrationsXml( request.getInputStream(), importOptions );
-            summary.setImportOptions( importOptions );
-            renderService.toXml( response.getOutputStream(), summary );
+            return asyncImport( importOptions, ImportCompleteDataSetRegistrationsTask.FORMAT_XML, request, response );
         }
+        response.setContentType( CONTENT_TYPE_XML );
+        ImportSummary summary = registrationExchangeService
+            .saveCompleteDataSetRegistrationsXml( request.getInputStream(), importOptions );
+        summary.setImportOptions( importOptions );
+        return WebMessageUtils.importSummary( summary );
```

Comment:
Obs! This is a change in behaviour. The `renderService.toXml( response.getOutputStream(), summary )` would only return the bare summary while this wraps the summary in the `WebMessage` envelope. This is needed to return a consistent type of result.